rasa/core/channels/twilio_voice.py

```diff
@@ -0,0 +1,116 @@
+            call_sid = request.form.get("CallSid")  # Get the call identifier
+
+            collector = TwilioVoiceCollectingOutputChannel()
+
+            # Provide an initial greeting to answer the user's call.
+            if (text is None) & (call_status == "ringing"):
+                text = "hello"
+
+            # determine the response.
+            if text is not None:
+                await on_new_message(
+                    UserMessage(
+                        text, collector, sender_id, input_channel=input_channel,
+                    )
+                )
+
+                # Parse the text responses and build the Twilio VoiceResponse.
+                respond_segments = []
+                for message in collector.messages:
+                    respond_segments.append(message["text"])
+                    if "buttons" in message:
+                        for button in message["buttons"]:
+                            respond_segments.append(button["title"])
+                twilio_response = build_twilio_voice_response(respond_segments)
+                return response.text(str(twilio_response), content_type="text/xml")
+            # If the user doesn't respond to the previous message resend the last message.
+            elif text is None:
+                # Get last user utterance from tracker.
+                tracker = request.app.agent.tracker_store.retrieve(sender_id)
+                last_response = next(
+                    (e for e in reversed(tracker.events) if isinstance(e, BotUttered)),
+                    None,
+                )
+
+                # If no previous utterance found say something generic.
+                if last_response is None:
+                    last_response = "I didn't get that."
+                else:
+                    last_response = last_response.text
+
+                twilio_response = build_twilio_voice_response([last_response])
+                return response.text(str(twilio_response), content_type="text/xml")
+
+        return twilio_voice_webhook
+
+
+class TwilioVoiceCollectingOutputChannel(CollectingOutputChannel):
+    """Output channel that collects send messages in a list
+    (doesn't send them anywhere, just collects them)."""
+
+    @classmethod
+    def name(cls) -> Text:
+        return "twilio_voice"
+
+
+def build_twilio_voice_response(messages: List[Text]) -> VoiceResponse:
+    """Builds the Twilio Voice Response object."""
+    vr = VoiceResponse()
+    gather = Gather(
+        input="speech",
+        action="/webhooks/twilio_voice/webhook",
+        actionOnEmptyResult=True,
+        speechTimeout="auto",
+    )
+
+    # Add pauses between messages.
+    # Add a listener to the last message to listen for user response.
+    for i, message in enumerate(messages):
+        if i + 1 == len(messages):
+            gather.say(message)
+            vr.append(gather)
+        else:
+            vr.say(message)
+            vr.pause(length=1)
```

Comment:
should the length of the pause be configurable?

Reply:
Having a pause of 1 seems like a sensible default. This could definitely be configured going forward if there is a need.